app/lib/shared/analyzer_memcache.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:gcloud/service_scope.dart' as ss;
+import 'package:logging/logging.dart';
+import 'package:memcache/memcache.dart';
+
+final Logger _logger = new Logger('pub.analyzer_memcache');
+
+/// Sets the analyzer memcache.
+void registerAnalyzerMemcache(AnalyzerMemcache value) =>
+    ss.register(#_analyzerMemcache, value);
+
+/// The active analyzer memcache.
+AnalyzerMemcache get analyzerMemcache => ss.lookup(#_analyzerMemcache);
+
+class AnalyzerMemcache {
+  final Memcache _memcache;
+  final String _prefix = 'dart_analyzer_api_';
+  final Duration _expiration = new Duration(minutes: 60);
```

Comment:
Maybe we should keep this in sync with `AppEnginePackageMemcache`. I have reduced the timeout back then from 60 to 10 minutes after using flexible runtimes with a per-instance memcache.

Now that we have a shared memcache (again), we should increase the other timeout also to 60 minutes again. 

If everything works, then the memcache invalidation will take care of taking the value out, once a new upload happened.

Reply:
Will do in a follow-up. I think more than half of the memcache-related code could be shared, I just wanted to put this out first.